pkg/reconciler/service/action.go

```diff
@@ -10,11 +13,15 @@ import (
 )
 
 type ActionContext struct {
-	KubeClient       kubernetes.Client
-	WorkspaceFactory *workspace.Factory
-	Context          context.Context
-	Logger           *zap.SugaredLogger
-	ChartProvider    *chart.Provider
+	KubeClient         kubernetes.Client
+	WorkspaceFactory   *workspace.Factory
+	Context            context.Context
+	Logger             *zap.SugaredLogger
+	ChartProvider      *chart.Provider
+	Kubeconfig         string
+	InClusterClientSet k8s.Interface
+	ConfigsMap         map[string]interface{}
```

Comment:
This should not go into the ActionContext as it's component specific. The context is intended to give component reconcilers access to cross-cutting features of the reconciler.

Reply:
To be handled on separate branch.

Reply:
Refactored.